ehrql/tables/beta/tpp.py

```diff
@@ -251,60 +251,142 @@ class ec_cost(EventFrame):
     )
     grand_total_payment_mff = Series(
         float,
-        description="TODO",
+        description="The total payment for the activity (Net_SLA_Payment + Tariff_MFF_Payment).",
     )
     tariff_total_payment = Series(
         float,
-        description="TODO",
+        description="The total payment according to the national tariff.",
     )
     arrival_date = Series(
         datetime.date,
-        description="TODO",
+        description="The date the patient self presented at the accident & emergency department, or arrived in an ambulance at the accident & emergency department.",
     )
     ec_decision_to_admit_date = Series(
         datetime.date,
-        description="TODO",
+        description="The date a decision to admit was made.",
     )
     ec_injury_date = Series(
         datetime.date,
-        description="TODO",
+        description="The date the patient was injured.",
     )
 
 
 @table
 class emergency_care_attendances(EventFrame):
-    id = Series(int)  # noqa: A003
-    arrival_date = Series(datetime.date)
-    discharge_destination = Series(SNOMEDCTCode)
+    id = Series(  # noqa: A003
+        int,
+        constraints=[Constraint.NotNull()],
+        description="Primary key used to join to the other EC tables",
+    )
+    arrival_date = Series(
+        datetime.date,
+        description="The date the patient self presented at the accident & emergency department, or arrived in an ambulance at the accident & emergency department.",
+    )
+    discharge_destination = Series(
+        SNOMEDCTCode,
+        description="The SNOMED CT concept ID which is used to identify the intended destination of the patient following discharge from the emergency care department.",
+    )
     # TODO: Revisit this when we have support for multi-valued fields
-    diagnosis_01 = Series(SNOMEDCTCode)
-    diagnosis_02 = Series(SNOMEDCTCode)
-    diagnosis_03 = Series(SNOMEDCTCode)
-    diagnosis_04 = Series(SNOMEDCTCode)
-    diagnosis_05 = Series(SNOMEDCTCode)
-    diagnosis_06 = Series(SNOMEDCTCode)
-    diagnosis_07 = Series(SNOMEDCTCode)
-    diagnosis_08 = Series(SNOMEDCTCode)
-    diagnosis_09 = Series(SNOMEDCTCode)
-    diagnosis_10 = Series(SNOMEDCTCode)
-    diagnosis_11 = Series(SNOMEDCTCode)
-    diagnosis_12 = Series(SNOMEDCTCode)
-    diagnosis_13 = Series(SNOMEDCTCode)
-    diagnosis_14 = Series(SNOMEDCTCode)
-    diagnosis_15 = Series(SNOMEDCTCode)
-    diagnosis_16 = Series(SNOMEDCTCode)
-    diagnosis_17 = Series(SNOMEDCTCode)
-    diagnosis_18 = Series(SNOMEDCTCode)
-    diagnosis_19 = Series(SNOMEDCTCode)
-    diagnosis_20 = Series(SNOMEDCTCode)
-    diagnosis_21 = Series(SNOMEDCTCode)
-    diagnosis_22 = Series(SNOMEDCTCode)
-    diagnosis_23 = Series(SNOMEDCTCode)
-    diagnosis_24 = Series(SNOMEDCTCode)
+    diagnosis_01 = Series(
+        SNOMEDCTCode,
+        description="The SNOMED CT concept ID which is used to identify the patient diagnosis.",
```

Comment:
question: Should this repetition be removed somehow? Or is it fine? It's the first collection of columns in this file that has an identical representation except for attribute names.

Reply:
Yeah, it's just felt very icky.